### PR TITLE
chore(lint): switch from tslint to eslint

### PR DESCRIPTION
**Pull request by wyze**
Upgrade to latest ESLint and install the typescript parser for ESLint. Using `eslint-config-react-app` so that is where all the plugins are coming from.

Removed `fmt-check` from CI as it is included in linting now.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,7 +34,5 @@ jobs:
       - run: yarn install --frozen-lockfile
       - run: make bootstrap
       - run: make packages-build
-      - run: make fmt-check
```
Comment (plouc) [inline .github/workflows/ci.yml:37]: `eslint-config-prettier` purpose is:

> Turns off all rules that are unnecessary or might conflict with Prettier.

So I think we still need to run `fmt-check` which serves a different purpose.

File: Makefile [hunk 2]
```diff
@@ -117,37 +117,7 @@ package-lint-%: ##@1 packages run eslint on package
 
 packages-lint: ##@1 packages run eslint on all packages
 	@echo "${YELLOW}Running eslint on all packages${RESET}"
-	@./node_modules/.bin/eslint "./packages/*/{src,tests}/**/*.js"
```
Comment (plouc) [inline Makefile:120]: nice ;)

**Review by plouc — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (wyze) [reply .github/workflows/ci.yml:37]: Are you okay adding eslint-plugin-prettier? I think I meant to do that but missed it.

Comment (plouc) [reply .github/workflows/ci.yml:37]: Oh, I wasn't aware of this plugin, I think it's good to be able to run everything with one command, I just hope we'll understand clearly when it's just about re-running prettier VS issues which have to be fixed manually. Let's add it and see how it goes.

**Review by plouc — APPROVED** [review]
LGTM

**Pull request merged**